diff --git a/.github/workflows/build-docker.yml b/.github/workflows/build-docker.yml
index cfa46b4f1..41ff52d39 100644
--- a/.github/workflows/build-docker.yml
+++ b/.github/workflows/build-docker.yml
@@ -48,7 +48,7 @@ jobs:
         uses: gradle/actions/setup-gradle@0723195856401067f7a2779048b490ace7a47d7c # v5.0.2
 
       - name: Build the image
-        run: ./gradlew --no-daemon --console=plain :fineract-provider:jibDockerBuild -Djib.to.image=$IMAGE_NAME -x test -x cucumber -x buildJavaSdk
+        run: ./gradlew --no-daemon --console=plain :fineract-war:jibDockerBuild -Djib.to.image=$IMAGE_NAME -x test -x cucumber -x buildJavaSdk
 
       - name: Verify git.properties in image
         run: |
diff --git a/.github/workflows/build-postgresql.yml b/.github/workflows/build-postgresql.yml
index 4cc23d091..c94dfdc5d 100644
--- a/.github/workflows/build-postgresql.yml
+++ b/.github/workflows/build-postgresql.yml
@@ -58,7 +58,7 @@ jobs:
           validate-wrappers: true
 
       - name: Build the image
-        run: ./gradlew --no-daemon --console=plain :fineract-provider:jibDockerBuild -Djib.to.image=$IMAGE_NAME -x test -x cucumber -x buildJavaSdk
+        run: ./gradlew --no-daemon --console=plain :fineract-war:war -x test -x cucumber -x buildJavaSdk
 
       - name: Start dependencies
         run: docker compose -f "$COMPOSE_FILE" up -d db localstack mock-oauth2-server
diff --git a/.github/workflows/publish-dockerhub.yml b/.github/workflows/publish-dockerhub.yml
index 95f241c41..5ae3c08b6 100644
--- a/.github/workflows/publish-dockerhub.yml
+++ b/.github/workflows/publish-dockerhub.yml
@@ -66,7 +66,7 @@ jobs:
               exit 0
             fi
           fi
-          ./gradlew --no-daemon --console=plain :fineract-provider:jib -x test -x cucumber -x buildJavaSdk  \
+          ./gradlew --no-daemon --console=plain :fineract-war:jib -x test -x cucumber -x buildJavaSdk  \
             -Djib.to.auth.username=${{secrets.DOCKERHUB_USER}} \
             -Djib.to.auth.password=${{secrets.DOCKERHUB_TOKEN}} \
             -Djib.from.platforms=linux/amd64,linux/arm64 \
diff --git a/build.gradle b/build.gradle
index 6cfc947e9..9011fd188 100644
--- a/build.gradle
+++ b/build.gradle
@@ -36,7 +36,6 @@ buildscript {
                 'fineract-command-async',
                 'fineract-command-disruptor',
                 'fineract-accounting',
-                'fineract-provider',
                 'fineract-branch',
                 'fineract-document',
                 'fineract-investor',
@@ -74,7 +73,6 @@ buildscript {
                 'fineract-command-jdbc',
                 'fineract-command-audit',
                 'fineract-accounting',
-                'fineract-provider',
                 'fineract-investor',
                 'fineract-charge',
                 'fineract-rates',
diff --git a/fineract-war/build.gradle b/fineract-war/build.gradle
index b7f5d9197..0281db732 100644
--- a/fineract-war/build.gradle
+++ b/fineract-war/build.gradle
@@ -20,8 +20,63 @@ description = 'Fineract WAR'
 
 apply plugin: 'war'
 apply plugin: 'distribution'
+apply plugin: 'org.springframework.boot'
+apply plugin: 'com.google.cloud.tools.jib'
 apply from: "${rootDir}/buildSrc/src/main/groovy/org.apache.fineract.dependencies.gradle"
 
+springBoot {
+    mainClass = 'org.apache.fineract.ServerApplication'
+}
+
+bootJar {
+    duplicatesStrategy = DuplicatesStrategy.EXCLUDE
+    manifest {
+        attributes('Main-Class': 'org.springframework.boot.loader.launch.PropertiesLauncher', 'Implementation-Title': 'Apache Fineract', 'Implementation-Version': project.version)
+    }
+    archiveClassifier = ''
+}
+
+jib {
+    from {
+        image = 'azul/zulu-openjdk-alpine:21'
+        platforms {
+            platform {
+                architecture = System.getProperty("os.arch").equals("aarch64")?"arm64":"amd64"
+                os = 'linux'
+            }
+        }
+    }
+
+    to {
+        image = 'fineract'
+        tags = [
+            "${project.version}",
+            'latest'
+        ]
+    }
+
+    container {
+        creationTime = 'USE_CURRENT_TIMESTAMP'
+        mainClass = 'org.apache.fineract.ServerApplication'
+        extraClasspath = ['/app/plugins/*']
+        args = [
+            '-Duser.home=/tmp',
+            '-Dfile.encoding=UTF-8',
+            '-Duser.timezone=UTC',
+            '-Djava.security.egd=file:/dev/./urandom'
+        ]
+        ports = ['8080/tcp', '8443/tcp']
+        labels = [maintainer: 'Aleksandar Vidakovic <aleks@apache.org>']
+        user = 'nobody:nogroup'
+    }
+
+    allowInsecureRegistries = true
+
+    dependencies {
+        implementation 'org.postgresql:postgresql'
+    }
+}
+
 war {
     description = 'Assembles a WAR archive containing the web application'
     archiveFileName = 'fineract-provider.war'
@@ -36,13 +91,13 @@ war {
     inputs.files(project.sourceSets.main.output.classesDirs)
         .withPropertyName("classes")
         .withPathSensitivity(PathSensitivity.RELATIVE)
-        
+    
     def licensesDir = file("$rootDir/licenses/binary/")
     if (licensesDir.exists()) {
         inputs.dir(licensesDir)
             .withPropertyName("licenses")
             .withPathSensitivity(PathSensitivity.RELATIVE)
-            
+        
         from(licensesDir) {
             into "WEB-INF/licenses/binary/"
         }
@@ -90,8 +145,186 @@ dependencies {
     implementation project(':fineract-loan-origination')
     implementation project(':fineract-progressive-loan')
     implementation project(':fineract-savings')
-    implementation project(':fineract-provider')
     implementation project(':fineract-working-capital-loan')
+    
+    // 包含原来fineract-provider的所有依赖
+    implementation(project(path: ':fineract-core'))
+    implementation(project(path: ':fineract-cob'))
+    implementation(project(path: ':fineract-command'))
+    implementation(project(path: ':fineract-command-audit'))
+    implementation(project(path: ':fineract-command-jdbc'))
+    implementation(project(path: ':fineract-validation'))
+    implementation(project(path: ':fineract-accounting'))
+    implementation(project(path: ':fineract-investor'))
+    implementation(project(path: ':fineract-rates'))
+    implementation(project(path: ':fineract-charge'))
+    implementation(project(path: ':fineract-loan'))
+    implementation(project(path: ':fineract-savings'))
+    implementation(project(path: ':fineract-branch'))
+    implementation(project(path: ':fineract-document'))
+    implementation(project(path: ':fineract-progressive-loan'))
+    implementation(project(path: ':fineract-report'))
+    implementation(project(path: ':fineract-tax'))
+    implementation(project(path: ':fineract-loan-origination'))
+    implementation(project(path: ':fineract-security'))
+    implementation(project(path: ':fineract-working-capital-loan'))
+    implementation(project(path: ':fineract-mix'))
+    
+    providedRuntime("org.springframework.boot:spring-boot-starter-tomcat")
+    
+    implementation(
+            project(path: ':fineract-avro-schemas')
+            )
+    implementation(
+            'org.springframework.boot:spring-boot-starter-web',
+            'org.springframework.boot:spring-boot-starter-validation',
+            'org.springframework.boot:spring-boot-starter-security',
+            'org.springframework.boot:spring-boot-starter-oauth2-authorization-server',
+            'org.springframework.boot:spring-boot-starter-cache',
+            'org.springframework.boot:spring-boot-starter-oauth2-resource-server',
+            'org.springframework.boot:spring-boot-starter-actuator',
+            'org.springframework.boot:spring-boot-starter-batch',
+            'org.springframework.boot:spring-boot-starter-data-jdbc',
+            'org.springframework.batch:spring-batch-integration',
+            'org.springframework.boot:spring-boot-starter-integration',
+            'org.springframework.integration:spring-integration-event',
+            'org.springframework.integration:spring-integration-jms',
+            'org.springframework.integration:spring-integration-kafka',
+            'org.springframework.kafka:spring-kafka',
+
+            'io.micrometer:micrometer-registry-prometheus',
+            'io.micrometer:micrometer-registry-otlp',
+            'io.micrometer:micrometer-registry-cloudwatch2',
+
+            'io.awspring.cloud:spring-cloud-aws-starter',
+
+            'org.apache.avro:avro',
+            'jakarta.ws.rs:jakarta.ws.rs-api',
+            'org.glassfish.jersey.media:jersey-media-multipart',
+
+            'org.springframework:spring-jms',
+            'org.springframework:spring-context-support',
+
+            'com.google.guava:guava',
+            'com.google.code.gson:gson',
+
+            'jakarta.jms:jakarta.jms-api',
+            'jakarta.management.j2ee:jakarta.management.j2ee-api',
+
+            'com.squareup.retrofit2:retrofit',
+            'com.squareup.okhttp3:okhttp',
+            'com.squareup.okhttp3:okhttp-urlconnection',
+
+            'org.apache.commons:commons-lang3',
+            'commons-io:commons-io',
+            'org.apache.poi:poi',
+            'org.apache.tika:tika-core',
+            'org.apache.tika:tika-parser-microsoft-module',
+            'org.apache.tika:tika-parser-miscoffice-module',
+
+            'org.liquibase:liquibase-core',
+
+            'com.github.librepdf:openpdf',
+            'com.github.spullara.mustache.java:compiler',
+            'com.jayway.jsonpath:json-path',
+
+            'javax.cache:cache-api',
+
+            'io.swagger.core.v3:swagger-annotations-jakarta',
+            'com.squareup.retrofit2:converter-gson',
+            'com.sun.activation:jakarta.activation',
+            'com.sun.mail:jakarta.mail',
+
+            'org.springdoc:springdoc-openapi-starter-webmvc-ui',
+            'org.mapstruct:mapstruct',
+
+            'io.github.resilience4j:resilience4j-spring-boot3',
+            'org.reflections:reflections',
+            )
+
+    implementation 'software.amazon.msk:aws-msk-iam-auth'
+
+    implementation 'io.swagger.core.v3:swagger-jaxrs2-jakarta'
+
+    implementation ('org.apache.commons:commons-email') {
+        exclude group: 'com.sun.mail', module: 'javax.mail'
+        exclude group: 'javax.activation', module: 'activation'
+    }
+
+    implementation('org.springframework.boot:spring-boot-starter-jersey') {
+        exclude group: 'org.glassfish.hk2.external', module: 'aopalliance-repackaged'
+        exclude group: 'org.glassfish.hk2', module: 'hk2-runlevel'
+        exclude group: 'org.hibernate.validator', module: 'hibernate-validator'
+        exclude group: 'jakarta.activation', module: 'jakarta.activation-api'
+    }
+    implementation('org.dom4j:dom4j') {
+        exclude group: 'javax.xml.bind'
+    }
+    implementation ('jakarta.xml.bind:jakarta.xml.bind-api') {
+        exclude group: 'jakarta.activation'
+    }
+    implementation ('org.apache.activemq:activemq-client') {
+        exclude group: 'org.apache.geronimo.specs'
+        exclude group: 'javax.annotation', module: 'javax.annotation-api'
+    }
+    implementation ('org.springframework.boot:spring-boot-starter-data-jpa') {
+        exclude group: 'org.hibernate'
+    }
+    implementation('org.eclipse.persistence:org.eclipse.persistence.jpa') {
+        exclude group: 'org.eclipse.persistence', module: 'jakarta.persistence'
+    }
+    implementation ('org.quartz-scheduler:quartz') {
+        exclude group: 'com.zaxxer', module: 'HikariCP-java7'
+    }
+    implementation('org.apache.commons:commons-csv'){}
+    implementation('org.apache.poi:poi-ooxml') {
+        exclude group: 'org.apache.poi', module: 'poi-ooxml-lite'
+    }
+    implementation ('software.amazon.awssdk:s3') {
+    }
+    implementation ('software.amazon.awssdk:auth') {
+    }
+
+    implementation ('org.mnode.ical4j:ical4j') {
+        exclude group: 'commons-logging'
+        exclude group: 'javax.activation'
+        exclude group: 'com.sun.mail', module: 'javax.mail'
+    }
+
+    implementation ('org.ehcache:ehcache') {
+        exclude group: 'com.sun.activation'
+    }
+
+    implementation('org.owasp.esapi:esapi') {
+        exclude group: 'org.slf4j'
+        exclude group: 'log4j'
+        exclude group: 'commons-logging'
+        exclude group: 'commons-collections'
+        exclude group: 'commons-lang'
+        exclude group: 'spotbugs-annotations'
+        exclude group: 'xml-apis'
+    }
+
+    runtimeOnly('org.glassfish.jaxb:jaxb-runtime') {
+        exclude group: 'com.sun.activation'
+    }
+
+    implementation('org.apache.httpcomponents:httpclient')
+
+    compileOnly 'com.github.spotbugs:spotbugs-annotations'
+    testCompileOnly 'com.github.spotbugs:spotbugs-annotations'
+    compileOnly 'org.projectlombok:lombok'
+    annotationProcessor 'org.projectlombok:lombok'
+    annotationProcessor 'org.mapstruct:mapstruct-processor'
+
+    implementation 'ch.qos.logback.contrib:logback-json-classic'
+    implementation 'ch.qos.logback.contrib:logback-jackson'
+    implementation 'org.codehaus.janino:janino'
+
+    implementation 'org.apache.commons:commons-math3'
+
+    implementation 'io.github.classgraph:classgraph'
+    implementation 'org.springframework.boot:spring-boot-starter-thymeleaf'
 }
 
 tasks.withType(Tar) {
@@ -121,10 +354,6 @@ distributions {
             from ("$rootDir/fineract-avro-schemas/build/libs/") {
                 include 'fineract-avro-schemas-*.jar'
             }
-            from ("$rootDir/fineract-provider/build/libs/") {
-                include 'fineract-provider-*.jar'
-                exclude 'fineract-provider-*-plain.jar'
-            }
             from war
             from("$rootDir/licenses/") {
                 into "licenses/"
@@ -174,7 +403,7 @@ tasks.named('binaryDistTar') {
     
     // Track dependencies explicitly
     dependsOn(war, ':fineract-client:jar', ':fineract-avro-schemas:jar', 
-              ':fineract-provider:build', ':fineract-doc:doc',
+              ':fineract-doc:doc',
               ':fineract-client:javadocJar', ':fineract-client:sourcesJar',
               ':fineract-avro-schemas:javadocJar', ':fineract-avro-schemas:sourcesJar')
 }
diff --git a/gradle/libs.versions.toml b/gradle/libs.versions.toml
new file mode 100644
index 000000000..85c5a8f95
--- /dev/null
+++ b/gradle/libs.versions.toml
@@ -0,0 +1,488 @@
+[versions]
+spring-boot = "3.5.14"
+jacoco = "0.8.14"
+junit = "5.14.4"
+junit-platform = "1.14.4"
+mockito = "5.23.0"
+lombok = "1.18.46"
+springdoc = "2.8.17"
+mapstruct = "1.6.3"
+swagger = "2.2.49"
+hibernate-validator = "9.1.0.Final"
+liquibase = "4.33.0"
+dom4j = "2.2.0"
+joda-time = "2.14.2"
+classgraph = "4.8.184"
+awaitility = "4.3.0"
+spotbugs-annotations = "4.9.8"
+jaxb-runtime = "2.3.6"
+jaxb-api = "4.0.5"
+jakarta-annotation = "3.0.0"
+jakarta-activation = "2.1.4"
+jakarta-mail = "2.0.2"
+jakarta-validation = "3.1.1"
+jakarta-jms = "3.1.0"
+jakarta-j2ee = "1.1.4"
+jakarta-ws-rs = "3.1.0"
+tomcat-embed = "10.1.55"
+thymeleaf = "3.1.5.RELEASE"
+netty = "4.1.133.Final"
+spring = "6.2.18"
+spring-security = "6.5.10"
+spring-kafka = ""
+spring-batch = ""
+spring-integration = ""
+spring-cloud-aws = "4.0.2"
+micrometer = "1.16.5"
+okhttp = "4.12.0"
+retrofit = "2.12.0"
+gson = "2.14.0"
+guava = "33.6.0-jre"
+commons-lang3 = "3.20.0"
+commons-collections4 = "4.5.0"
+commons-compress = "1.28.0"
+commons-io = "2.22.0"
+commons-codec = "1.22.0"
+commons-csv = "1.14.1"
+commons-math3 = "3.6.1"
+commons-email = "1.6.0"
+poi = "5.5.1"
+tika = "3.3.0"
+ical4j = "3.2.19"
+quartz = "2.5.2"
+ehcache = "3.12.0"
+mustache = "0.9.14"
+json-path = "3.0.0"
+librepdf = "3.0.4"
+esapi = "2.7.0.0"
+httpclient = "4.5.14"
+httpcore5 = "5.4"
+activemq = "6.2.5"
+aws-msk-iam-auth = "2.3.6"
+awssdk = "2.44.4"
+resilience4j = "2.4.0"
+reflections = "0.10.2"
+image-comparison = "4.4.0"
+icu4j = "78.3"
+spring-icu4j = "0.5.2"
+nimbus-jose-jwt = "10.9"
+jgit = "7.6.0.202603022253-r"
+svnkit = "1.10.13"
+semver4j = "3.1.0"
+text-io = "3.4.1"
+jul-to-slf4j = ""
+logback-classic = "1.5.32"
+logback-core = "1.5.32"
+logback-json = "0.1.5"
+logback-jackson = "0.1.5"
+slf4j = "2.0.17"
+janino = "3.1.12"
+jaxrs = "3.1.11"
+jersey-media-multipart = ""
+jersey = ""
+jersey-hk2 = ""
+eclipselink = "4.0.9"
+xml-apis = ""
+jsr305 = "3.0.2"
+groovy = "5.0.6"
+jmh = "1.37"
+jmh-annprocess = "1.37"
+jmh-core = "1.37"
+opencsv = ""
+rest-assured = "5.5.7"
+wiremock = "3.13.2"
+sshd-common = "2.17.1"
+sshd-core = "2.17.1"
+junit-platform-suite = "1.14.4"
+bouncycastle = "1.84"
+junit-jupiter-api = "5.14.4"
+junit-jupiter-engine = "5.14.4"
+junit-jupiter-params = "5.14.4"
+junit-platform-commons = "1.14.4"
+junit-platform-engine = "1.14.4"
+junit-platform-launcher = "1.14.4"
+junit-platform-suite-api = "1.14.4"
+junit-platform-suite-commons = "1.14.4"
+junit-platform-suite-engine = "1.14.4"
+mockito-inline = "5.2.0"
+mockito-junit-jupiter = "5.23.0"
+mockito-core = "5.23.0"
+mock-server = "5.14.0"
+rest-assured-json-path = "5.5.7"
+rest-assured-xml-path = "5.5.7"
+assertj-core = "3.27.7"
+greenmail = "2.0.1"
+webjars-locator-core = "0.59"
+avro = "1.12.1"
+mariadb-java-client = "3.5.8"
+postgresql = "42.7.11"
+mysql-connector-j = "9.7.0"
+lz4-java = "1.11.0"
+cucumber = "7.34.3"
+cucumber-java = "7.34.3"
+cucumber-java8 = "7.34.3"
+cucumber-junit-platform-engine = "7.34.3"
+cucumber-spring = "7.34.3"
+disruptor = "4.0.0"
+jackson = "2.21.3"
+kotlin = "2.3.21"
+opentelemetry = "1.62.0"
+testcontainers = "1.21.4"
+gson-fire = "1.9.0"
+cache-api = "1.1.1"
+jersey-media-multipart = ""
+jersey-core = ""
+jersey-server = ""
+jersey-client = ""
+spring-restdocs = "3.0.5"
+modernizer = ""
+gradle-cargo = "2.9.0"
+gradle-git-properties = "2.5.7"
+jib = "3.5.3"
+sonarqube = "6.3.1.5724"
+modernizer-plugin = "1.13.0"
+spotbugs-plugin = "6.5.4"
+cucumber-runner = "0.0.14"
+avro-plugin = "1.9.1"
+openapi-generator = "7.22.0"
+shadow = "9.4.1"
+jmh-plugin = "0.7.3"
+cyclonedx-bom = "3.2.4"
+swagger-brake = "2.7.0"
+git-versioning = "6.4.4"
+task-info = "2.2.1"
+test-logger = "4.0.0"
+spotless = "6.25.0"
+rat = "0.8.1"
+license = "0.16.1"
+dep-license-report = "2.9"
+jrebel = "1.2.2"
+errorprone = "4.4.0"
+swagger-gradle = "2.2.49"
+asciidoctor = "4.0.5"
+
+[libraries]
+spring-boot-web = { group = "org.springframework.boot", name = "spring-boot-starter-web", version.ref = "spring-boot" }
+spring-boot-security = { group = "org.springframework.boot", name = "spring-boot-starter-security", version.ref = "spring-boot" }
+spring-boot-validation = { group = "org.springframework.boot", name = "spring-boot-starter-validation", version.ref = "spring-boot" }
+spring-boot-cache = { group = "org.springframework.boot", name = "spring-boot-starter-cache", version.ref = "spring-boot" }
+spring-boot-actuator = { group = "org.springframework.boot", name = "spring-boot-starter-actuator", version.ref = "spring-boot" }
+spring-boot-batch = { group = "org.springframework.boot", name = "spring-boot-starter-batch", version.ref = "spring-boot" }
+spring-boot-data-jpa = { group = "org.springframework.boot", name = "spring-boot-starter-data-jpa", version.ref = "spring-boot" }
+spring-boot-data-jdbc = { group = "org.springframework.boot", name = "spring-boot-starter-data-jdbc", version.ref = "spring-boot" }
+spring-boot-jersey = { group = "org.springframework.boot", name = "spring-boot-starter-jersey", version.ref = "spring-boot" }
+spring-boot-test = { group = "org.springframework.boot", name = "spring-boot-starter-test", version.ref = "spring-boot" }
+spring-boot-oauth2-resource-server = { group = "org.springframework.boot", name = "spring-boot-starter-oauth2-resource-server", version.ref = "spring-boot" }
+spring-boot-oauth2-authorization-server = { group = "org.springframework.boot", name = "spring-boot-starter-oauth2-authorization-server", version.ref = "spring-boot" }
+spring-boot-thymeleaf = { group = "org.springframework.boot", name = "spring-boot-starter-thymeleaf", version.ref = "spring-boot" }
+spring-batch-integration = { group = "org.springframework.batch", name = "spring-batch-integration" }
+spring-integration-event = { group = "org.springframework.integration", name = "spring-integration-event" }
+spring-integration-jms = { group = "org.springframework.integration", name = "spring-integration-jms" }
+spring-integration-kafka = { group = "org.springframework.integration", name = "spring-integration-kafka" }
+spring-kafka = { group = "org.springframework.kafka", name = "spring-kafka" }
+spring-jms = { group = "org.springframework", name = "spring-jms", version.ref = "spring" }
+spring-context-support = { group = "org.springframework", name = "spring-context-support", version.ref = "spring" }
+spring-core = { group = "org.springframework", name = "spring-core", version.ref = "spring" }
+spring-security-core = { group = "org.springframework.security", name = "spring-security-core", version.ref = "spring-security" }
+springdoc-openapi-starter-webmvc-ui = { group = "org.springdoc", name = "springdoc-openapi-starter-webmvc-ui", version.ref = "springdoc" }
+
+micrometer-registry-prometheus = { group = "io.micrometer", name = "micrometer-registry-prometheus" }
+micrometer-registry-otlp = { group = "io.micrometer", name = "micrometer-registry-otlp" }
+micrometer-registry-cloudwatch2 = { group = "io.micrometer", name = "micrometer-registry-cloudwatch2" }
+
+spring-cloud-aws-starter = { group = "io.awspring.cloud", name = "spring-cloud-aws-starter", version.ref = "spring-cloud-aws" }
+
+commons-lang3 = { group = "org.apache.commons", name = "commons-lang3", version.ref = "commons-lang3" }
+commons-collections4 = { group = "org.apache.commons", name = "commons-collections4", version.ref = "commons-collections4" }
+commons-compress = { group = "org.apache.commons", name = "commons-compress", version.ref = "commons-compress" }
+commons-io = { group = "commons-io", name = "commons-io", version.ref = "commons-io" }
+commons-codec = { group = "commons-codec", name = "commons-codec", version.ref = "commons-codec" }
+commons-csv = { group = "org.apache.commons", name = "commons-csv", version.ref = "commons-csv" }
+commons-math3 = { group = "org.apache.commons", name = "commons-math3", version.ref = "commons-math3" }
+commons-email = { group = "org.apache.commons", name = "commons-email", version.ref = "commons-email" }
+
+gson = { group = "com.google.code.gson", name = "gson", version.ref = "gson" }
+guava = { group = "com.google.guava", name = "guava", version.ref = "guava" }
+
+logback-classic = { group = "ch.qos.logback", name = "logback-classic", version.ref = "logback-classic" }
+logback-core = { group = "ch.qos.logback", name = "logback-core", version.ref = "logback-core" }
+logback-json-classic = { group = "ch.qos.logback.contrib", name = "logback-json-classic", version.ref = "logback-json" }
+logback-jackson = { group = "ch.qos.logback.contrib", name = "logback-jackson", version.ref = "logback-jackson" }
+slf4j-api = { group = "org.slf4j", name = "slf4j-api", version.ref = "slf4j" }
+jul-to-slf4j = { group = "org.slf4j", name = "jul-to-slf4j", version.ref = "slf4j" }
+log4j-over-slf4j = { group = "org.slf4j", name = "log4j-over-slf4j", version.ref = "slf4j" }
+
+lombok = { group = "org.projectlombok", name = "lombok", version.ref = "lombok" }
+mapstruct = { group = "org.mapstruct", name = "mapstruct", version.ref = "mapstruct" }
+mapstruct-processor = { group = "org.mapstruct", name = "mapstruct-processor", version.ref = "mapstruct" }
+
+jackson-bom = { group = "com.fasterxml.jackson", name = "jackson-bom", version.ref = "jackson" }
+
+jackson-databind = { group = "com.fasterxml.jackson.core", name = "jackson-databind" }
+jackson-core = { group = "com.fasterxml.jackson.core", name = "jackson-core" }
+jackson-annotations = { group = "com.fasterxml.jackson.core", name = "jackson-annotations" }
+
+jaxb-api = { group = "jakarta.xml.bind", name = "jakarta.xml.bind-api", version.ref = "jaxb-api" }
+jaxb-runtime = { group = "org.glassfish.jaxb", name = "jaxb-runtime", version.ref = "jaxb-runtime" }
+jakarta-activation = { group = "jakarta.activation", name = "jakarta.activation-api", version.ref = "jakarta-activation" }
+jakarta-mail = { group = "com.sun.mail", name = "jakarta.mail", version.ref = "jakarta-mail" }
+jakarta-annotation-api = { group = "jakarta.annotation", name = "jakarta.annotation-api", version.ref = "jakarta-annotation" }
+jakarta-validation-api = { group = "jakarta.validation", name = "jakarta.validation-api", version.ref = "jakarta-validation" }
+jakarta-jms-api = { group = "jakarta.jms", name = "jakarta.jms-api", version.ref = "jakarta-jms" }
+jakarta-j2ee-api = { group = "jakarta.management.j2ee", name = "jakarta.management.j2ee-api", version.ref = "jakarta-j2ee" }
+jakarta-ws-rs-api = { group = "jakarta.ws.rs", name = "jakarta.ws.rs-api", version.ref = "jakarta-ws-rs" }
+
+hibernate-validator = { group = "org.hibernate.validator", name = "hibernate-validator", version.ref = "hibernate-validator" }
+
+eclipselink-jpa = { group = "org.eclipse.persistence", name = "org.eclipse.persistence.jpa", version.ref = "eclipselink" }
+
+thymeleaf = { group = "org.thymeleaf", name = "thymeleaf", version.ref = "thymeleaf" }
+thymeleaf-spring6 = { group = "org.thymeleaf", name = "thymeleaf-spring6", version.ref = "thymeleaf" }
+
+tomcat-embed-core = { group = "org.apache.tomcat.embed", name = "tomcat-embed-core", version.ref = "tomcat-embed" }
+tomcat-embed-el = { group = "org.apache.tomcat.embed", name = "tomcat-embed-el", version.ref = "tomcat-embed" }
+tomcat-embed-websocket = { group = "org.apache.tomcat.embed", name = "tomcat-embed-websocket", version.ref = "tomcat-embed" }
+
+netty-buffer = { group = "io.netty", name = "netty-buffer", version.ref = "netty" }
+netty-common = { group = "io.netty", name = "netty-common", version.ref = "netty" }
+netty-codec = { group = "io.netty", name = "netty-codec", version.ref = "netty" }
+netty-codec-base = { group = "io.netty", name = "netty-codec-base", version.ref = "netty" }
+netty-codec-compression = { group = "io.netty", name = "netty-codec-compression", version.ref = "netty" }
+netty-codec-http = { group = "io.netty", name = "netty-codec-http", version.ref = "netty" }
+netty-codec-http2 = { group = "io.netty", name = "netty-codec-http2", version.ref = "netty" }
+netty-codec-marshalling = { group = "io.netty", name = "netty-codec-marshalling", version.ref = "netty" }
+netty-codec-protobuf = { group = "io.netty", name = "netty-codec-protobuf", version.ref = "netty" }
+netty-codec-socks = { group = "io.netty", name = "netty-codec-socks", version.ref = "netty" }
+netty-handler = { group = "io.netty", name = "netty-handler", version.ref = "netty" }
+netty-handler-proxy = { group = "io.netty", name = "netty-handler-proxy", version.ref = "netty" }
+netty-resolver = { group = "io.netty", name = "netty-resolver", version.ref = "netty" }
+netty-transport = { group = "io.netty", name = "netty-transport", version.ref = "netty" }
+netty-transport-classes-epoll = { group = "io.netty", name = "netty-transport-classes-epoll", version.ref = "netty" }
+netty-transport-native-unix-common = { group = "io.netty", name = "netty-transport-native-unix-common", version.ref = "netty" }
+
+jersey-media-multipart = { group = "org.glassfish.jersey.media", name = "jersey-media-multipart" }
+jersey-core = { group = "org.glassfish.jersey.core", name = "jersey-core" }
+jersey-server = { group = "org.glassfish.jersey.core", name = "jersey-server" }
+jersey-client = { group = "org.glassfish.jersey.core", name = "jersey-client" }
+jersey-hk2 = { group = "org.glassfish.jersey.inject", name = "jersey-hk2" }
+
+swagger-annotations-jakarta = { group = "io.swagger.core.v3", name = "swagger-annotations-jakarta", version.ref = "swagger" }
+swagger-jaxrs2-jakarta = { group = "io.swagger.core.v3", name = "swagger-jaxrs2-jakarta", version.ref = "swagger" }
+swagger-core-jakarta = { group = "io.swagger.core.v3", name = "swagger-core-jakarta", version.ref = "swagger" }
+
+retrofit = { group = "com.squareup.retrofit2", name = "retrofit", version.ref = "retrofit" }
+retrofit-converter-gson = { group = "com.squareup.retrofit2", name = "converter-gson", version.ref = "retrofit" }
+retrofit-converter-simplexml = { group = "com.squareup.retrofit2", name = "converter-simplexml", version.ref = "retrofit" }
+retrofit-converter-jaxb = { group = "com.squareup.retrofit2", name = "converter-jaxb", version.ref = "retrofit" }
+retrofit-converter-java8 = { group = "com.squareup.retrofit2", name = "converter-java8", version.ref = "retrofit" }
+retrofit-converter-scalars = { group = "com.squareup.retrofit2", name = "converter-scalars", version.ref = "retrofit" }
+retrofit-converter-protobuf = { group = "com.squareup.retrofit2", name = "converter-protobuf", version.ref = "retrofit" }
+retrofit-converter-wire = { group = "com.squareup.retrofit2", name = "converter-wire", version.ref = "retrofit" }
+retrofit-adapter-rxjava2 = { group = "com.squareup.retrofit2", name = "adapter-rxjava2", version.ref = "retrofit" }
+retrofit-adapter-rxjava3 = { group = "com.squareup.retrofit2", name = "adapter-rxjava3", version.ref = "retrofit" }
+retrofit-adapter-guava = { group = "com.squareup.retrofit2", name = "adapter-guava", version.ref = "retrofit" }
+retrofit-adapter-java8 = { group = "com.squareup.retrofit2", name = "adapter-java8", version.ref = "retrofit" }
+retrofit-mock = { group = "com.squareup.retrofit2", name = "retrofit-mock", version.ref = "retrofit" }
+
+okhttp = { group = "com.squareup.okhttp3", name = "okhttp", version.ref = "okhttp" }
+okhttp-urlconnection = { group = "com.squareup.okhttp3", name = "okhttp-urlconnection", version.ref = "okhttp" }
+
+gson-fire = { group = "io.gsonfire", name = "gson-fire", version.ref = "gson-fire" }
+
+jsr305 = { group = "com.google.code.findbugs", name = "jsr305", version.ref = "jsr305" }
+
+quartz = { group = "org.quartz-scheduler", name = "quartz", version.ref = "quartz" }
+
+ical4j = { group = "org.mnode.ical4j", name = "ical4j", version.ref = "ical4j" }
+
+liquibase-core = { group = "org.liquibase", name = "liquibase-core", version.ref = "liquibase" }
+liquibase-postgresql = { group = "org.liquibase.ext", name = "liquibase-postgresql", version.ref = "liquibase" }
+
+dom4j = { group = "org.dom4j", name = "dom4j", version.ref = "dom4j" }
+xml-apis = { group = "xml-apis", name = "xml-apis", version.ref = "xml-apis" }
+
+joda-time = { group = "joda-time", name = "joda-time", version.ref = "joda-time" }
+
+classgraph = { group = "io.github.classgraph", name = "classgraph", version.ref = "classgraph" }
+
+awaitility = { group = "org.awaitility", name = "awaitility", version.ref = "awaitility" }
+
+spotbugs-annotations = { group = "com.github.spotbugs", name = "spotbugs-annotations", version.ref = "spotbugs-annotations" }
+
+cache-api = { group = "javax.cache", name = "cache-api", version.ref = "cache-api" }
+
+ehcache = { group = "org.ehcache", name = "ehcache", version.ref = "ehcache" }
+
+mustache = { group = "com.github.spullara.mustache.java", name = "compiler", version.ref = "mustache" }
+
+json-path = { group = "com.jayway.jsonpath", name = "json-path", version.ref = "json-path" }
+
+librepdf = { group = "com.github.librepdf", name = "openpdf", version.ref = "librepdf" }
+
+esapi = { group = "org.owasp.esapi", name = "esapi", version.ref = "esapi" }
+
+httpclient = { group = "org.apache.httpcomponents", name = "httpclient", version.ref = "httpclient" }
+httpcore5 = { group = "org.apache.httpcomponents.core5", name = "httpcore5", version.ref = "httpcore5" }
+httpcore5-h2 = { group = "org.apache.httpcomponents.core5", name = "httpcore5-h2", version.ref = "httpcore5" }
+
+activemq-client = { group = "org.apache.activemq", name = "activemq-client", version.ref = "activemq" }
+
+aws-msk-iam-auth = { group = "software.amazon.msk", name = "aws-msk-iam-auth", version.ref = "aws-msk-iam-auth" }
+
+awssdk-s3 = { group = "software.amazon.awssdk", name = "s3", version.ref = "awssdk" }
+awssdk-auth = { group = "software.amazon.awssdk", name = "auth", version.ref = "awssdk" }
+
+resilience4j-spring-boot3 = { group = "io.github.resilience4j", name = "resilience4j-spring-boot3", version.ref = "resilience4j" }
+resilience4j-core = { group = "io.github.resilience4j", name = "resilience4j-core", version.ref = "resilience4j" }
+resilience4j-circuitbreaker = { group = "io.github.resilience4j", name = "resilience4j-circuitbreaker", version.ref = "resilience4j" }
+
+reflections = { group = "org.reflections", name = "reflections", version.ref = "reflections" }
+
+image-comparison = { group = "com.github.romankh3", name = "image-comparison", version.ref = "image-comparison" }
+
+icu4j = { group = "com.ibm.icu", name = "icu4j", version.ref = "icu4j" }
+spring-icu4j = { group = "org.yakworks", name = "spring-icu4j", version.ref = "spring-icu4j" }
+
+nimbus-jose-jwt = { group = "com.nimbusds", name = "nimbus-jose-jwt", version.ref = "nimbus-jose-jwt" }
+
+jgit = { group = "org.eclipse.jgit", name = "org.eclipse.jgit", version.ref = "jgit" }
+jgit-gpg-bc = { group = "org.eclipse.jgit", name = "org.eclipse.jgit.gpg.bc", version.ref = "jgit" }
+jgit-ssh-apache = { group = "org.eclipse.jgit", name = "org.eclipse.jgit.ssh.apache", version.ref = "jgit" }
+
+svnkit = { group = "com.tmatesoft.svnkit", name = "svnkit", version.ref = "svnkit" }
+
+semver4j = { group = "com.vdurmont", name = "semver4j", version.ref = "semver4j" }
+
+text-io = { group = "org.beryx", name = "text-io", version.ref = "text-io" }
+
+janino = { group = "org.codehaus.janino", name = "janino", version.ref = "janino" }
+
+poi = { group = "org.apache.poi", name = "poi", version.ref = "poi" }
+poi-ooxml = { group = "org.apache.poi", name = "poi-ooxml", version.ref = "poi" }
+poi-ooxml-lite = { group = "org.apache.poi", name = "poi-ooxml-lite", version.ref = "poi" }
+poi-ooxml-schemas = { group = "org.apache.poi", name = "poi-ooxml-schemas", version.ref = "poi" }
+
+tika-core = { group = "org.apache.tika", name = "tika-core", version.ref = "tika" }
+tika-parser-microsoft-module = { group = "org.apache.tika", name = "tika-parser-microsoft-module", version.ref = "tika" }
+tika-parser-miscoffice-module = { group = "org.apache.tika", name = "tika-parser-miscoffice-module", version.ref = "tika" }
+
+avro = { group = "org.apache.avro", name = "avro", version.ref = "avro" }
+
+mariadb-java-client = { group = "org.mariadb.jdbc", name = "mariadb-java-client", version.ref = "mariadb-java-client" }
+postgresql = { group = "org.postgresql", name = "postgresql", version.ref = "postgresql" }
+mysql-connector-j = { group = "com.mysql", name = "mysql-connector-j", version.ref = "mysql-connector-j" }
+
+lz4-java = { group = "at.yawk.lz4", name = "lz4-java", version.ref = "lz4-java" }
+
+junit-jupiter-api = { group = "org.junit.jupiter", name = "junit-jupiter-api", version.ref = "junit-jupiter-api" }
+junit-jupiter-engine = { group = "org.junit.jupiter", name = "junit-jupiter-engine", version.ref = "junit-jupiter-engine" }
+junit-jupiter-params = { group = "org.junit.jupiter", name = "junit-jupiter-params", version.ref = "junit-jupiter-params" }
+junit-platform-commons = { group = "org.junit.platform", name = "junit-platform-commons", version.ref = "junit-platform" }
+junit-platform-engine = { group = "org.junit.platform", name = "junit-platform-engine", version.ref = "junit-platform" }
+junit-platform-launcher = { group = "org.junit.platform", name = "junit-platform-launcher", version.ref = "junit-platform" }
+junit-platform-suite = { group = "org.junit.platform", name = "junit-platform-suite", version.ref = "junit-platform-suite" }
+junit-platform-suite-api = { group = "org.junit.platform", name = "junit-platform-suite-api", version.ref = "junit-platform-suite-api" }
+junit-platform-suite-commons = { group = "org.junit.platform", name = "junit-platform-suite-commons", version.ref = "junit-platform-suite-commons" }
+junit-platform-suite-engine = { group = "org.junit.platform", name = "junit-platform-suite-engine", version.ref = "junit-platform-suite-engine" }
+
+mockito-core = { group = "org.mockito", name = "mockito-core", version.ref = "mockito-core" }
+mockito-junit-jupiter = { group = "org.mockito", name = "mockito-junit-jupiter", version.ref = "mockito-junit-jupiter" }
+mockito-inline = { group = "org.mockito", name = "mockito-inline", version.ref = "mockito-inline" }
+
+mock-server = { group = "org.mock-server", name = "mock-server-junit-jupiter", version.ref = "mock-server" }
+
+rest-assured = { group = "io.rest-assured", name = "rest-assured", version.ref = "rest-assured" }
+rest-assured-json-path = { group = "io.rest-assured", name = "json-path", version.ref = "rest-assured-json-path" }
+rest-assured-xml-path = { group = "io.rest-assured", name = "xml-path", version.ref = "rest-assured-xml-path" }
+
+assertj-core = { group = "org.assertj", name = "assertj-core", version.ref = "assertj-core" }
+
+greenmail = { group = "com.icegreen", name = "greenmail-junit5", version.ref = "greenmail" }
+
+webjars-locator-core = { group = "org.webjars", name = "webjars-locator-core", version.ref = "webjars-locator-core" }
+
+bouncycastle-bcpkix-jdk18on = { group = "org.bouncycastle", name = "bcpkix-jdk18on", version.ref = "bouncycastle" }
+bouncycastle-bcprov-jdk18on = { group = "org.bouncycastle", name = "bcprov-jdk18on", version.ref = "bouncycastle" }
+bouncycastle-bcutil-jdk18on = { group = "org.bouncycastle", name = "bcutil-jdk18on", version.ref = "bouncycastle" }
+bouncycastle-bcpg-jdk18on = { group = "org.bouncycastle", name = "bcpg-jdk18on", version.ref = "bouncycastle" }
+
+cucumber-java = { group = "io.cucumber", name = "cucumber-java", version.ref = "cucumber-java" }
+cucumber-java8 = { group = "io.cucumber", name = "cucumber-java8", version.ref = "cucumber-java8" }
+cucumber-junit-platform-engine = { group = "io.cucumber", name = "cucumber-junit-platform-engine", version.ref = "cucumber-junit-platform-engine" }
+cucumber-spring = { group = "io.cucumber", name = "cucumber-spring", version.ref = "cucumber-spring" }
+cucumber-core = { group = "io.cucumber", name = "cucumber-core", version.ref = "cucumber" }
+
+disruptor = { group = "com.lmax", name = "disruptor", version.ref = "disruptor" }
+
+groovy-xml = { group = "org.apache.groovy", name = "groovy-xml", version.ref = "groovy" }
+groovy-json = { group = "org.apache.groovy", name = "groovy-json", version.ref = "groovy" }
+
+jmh-core = { group = "org.openjdk.jmh", name = "jmh-core", version.ref = "jmh-core" }
+jmh-annprocess = { group = "org.openjdk.jmh", name = "jmh-generator-annprocess", version.ref = "jmh-annprocess" }
+
+spring-restdocs-asciidoctor = { group = "org.springframework.restdocs", name = "spring-restdocs-asciidoctor", version.ref = "spring-restdocs" }
+spring-restdocs-mockmvc = { group = "org.springframework.restdocs", name = "spring-restdocs-mockmvc", version.ref = "spring-restdocs" }
+spring-restdocs-webtestclient = { group = "org.springframework.restdocs", name = "spring-restdocs-webtestclient", version.ref = "spring-restdocs" }
+spring-restdocs-restassured = { group = "org.springframework.restdocs", name = "spring-restdocs-restassured", version.ref = "spring-restdocs" }
+
+wiremock = { group = "org.wiremock", name = "wiremock-standalone", version.ref = "wiremock" }
+
+sshd-common = { group = "org.apache.sshd", name = "sshd-common", version.ref = "sshd-common" }
+sshd-core = { group = "org.apache.sshd", name = "sshd-core", version.ref = "sshd-core" }
+
+rxjava2 = { group = "io.reactivex.rxjava2", name = "rxjava", version.ref = "rxjava2" }
+
+rxjava3 = { group = "io.reactivex.rxjava3", name = "rxjava", version.ref = "rxjava3" }
+
+rxjava2 = "2.2.21"
+rxjava3 = ""
+kotlin-bom = { group = "org.jetbrains.kotlin", name = "kotlin-bom", version.ref = "kotlin" }
+kotlin-stdlib = { group = "org.jetbrains.kotlin", name = "kotlin-stdlib", version.ref = "kotlin" }
+kotlin-stdlib-jdk8 = { group = "org.jetbrains.kotlin", name = "kotlin-stdlib-jdk8", version.ref = "kotlin" }
+
+opentelemetry-bom = { group = "io.opentelemetry", name = "opentelemetry-bom", version.ref = "opentelemetry" }
+
+testcontainers-bom = { group = "org.testcontainers", name = "testcontainers-bom", version.ref = "testcontainers" }
+
+okhttp-bom = { group = "com.squareup.okhttp3", name = "okhttp-bom", version.ref = "okhttp" }
+slf4j-bom = { group = "org.slf4j", name = "slf4j-bom", version.ref = "slf4j" }
+micrometer-bom = { group = "io.micrometer", name = "micrometer-bom", version.ref = "micrometer" }
+spring-boot-bom = { group = "org.springframework.boot", name = "spring-boot-dependencies", version.ref = "spring-boot" }
+spring-cloud-aws-bom = { group = "io.awspring.cloud", name = "spring-cloud-aws-dependencies", version.ref = "spring-cloud-aws" }
+junit-bom = { group = "org.junit", name = "junit-bom", version.ref = "junit" }
+jackson-bom2 = { group = "com.fasterxml.jackson", name = "jackson-bom", version.ref = "jackson" }
+cucumber-bom = { group = "io.cucumber", name = "cucumber-bom", version.ref = "cucumber" }
+mockito-bom = { group = "org.mockito", name = "mockito-bom", version.ref = "mockito" }
+awssdk-bom = { group = "software.amazon.awssdk", name = "bom", version.ref = "awssdk" }
+resilience4j-bom = { group = "io.github.resilience4j", name = "resilience4j-bom", version.ref = "resilience4j" }
+testcontainers-bom2 = { group = "org.testcontainers", name = "testcontainers-bom", version.ref = "testcontainers" }
+jersey-bom = { group = "org.glassfish.jersey", name = "jersey-bom", version.ref = "jersey" }
+
+[plugins]
+spring-boot = { id = "org.springframework.boot", version.ref = "spring-boot" }
+io-spring-dependency-management = { id = "io.spring.dependency-management", version = "1.1.4" }
+com-google-cloud-tools-jib = { id = "com.google.cloud.tools.jib", version.ref = "jib" }
+org-sonarqube = { id = "org.sonarqube", version.ref = "sonarqube" }
+com-github-spotbugs = { id = "com.github.spotbugs", version.ref = "spotbugs-plugin" }
+se-thinkcode-cucumber-runner = { id = "se.thinkcode.cucumber-runner", version.ref = "cucumber-runner" }
+com-github-gradle-up-shadow = { id = "com.gradleup.shadow", version.ref = "shadow" }
+me-champeau-jmh = { id = "me.champeau.jmh", version.ref = "jmh-plugin" }
+org-cyclonedx-bom = { id = "org.cyclonedx.bom", version.ref = "cyclonedx-bom" }
+com-docktape-swagger-brake = { id = "com.docktape.swagger-brake", version.ref = "swagger-brake" }
+com-diffplug-spotless = { id = "com.diffplug.spotless", version.ref = "spotless" }
+org-nosphere-apache-rat = { id = "org.nosphere.apache.rat", version.ref = "rat" }
+com-github-hierynomus-license = { id = "com.github.hierynomus.license", version.ref = "license" }
+com-github-jk1-dependency-license-report = { id = "com.github.jk1.dependency-license-report", version.ref = "dep-license-report" }
+org-zeroturnaround-gradle-jrebel = { id = "org.zeroturnaround.gradle.jrebel", version.ref = "jrebel" }
+net-ltgt-errorprone = { id = "net.ltgt.errorprone", version.ref = "errorprone" }
+io-swagger-core-v3-swagger-gradle-plugin = { id = "io.swagger.core.v3.swagger-gradle-plugin", version.ref = "swagger-gradle" }
+com-gorylenko-gradle-git-properties = { id = "com.gorylenko.gradle-git-properties", version.ref = "gradle-git-properties" }
+org-asciidoctor-jvm-convert = { id = "org.asciidoctor.jvm.convert", version.ref = "asciidoctor" }
+org-asciidoctor-jvm-pdf = { id = "org.asciidoctor.jvm.pdf", version.ref = "asciidoctor" }
+com-github-andygoossens-modernizer = { id = "com.github.andygoossens.modernizer", version.ref = "modernizer-plugin" }
+com-github-davidmc24-gradle-plugin-avro-base = { id = "com.github.davidmc24.gradle.plugin.avro-base", version.ref = "avro-plugin" }
+org-openapi-generator = { id = "org.openapi.generator", version.ref = "openapi-generator" }
+me-qoomon-git-versioning = { id = "me.qoomon.git-versioning", version.ref = "git-versioning" }
+org-barfuin-gradle-task-info = { id = "org.barfuin.gradle.taskinfo", version.ref = "task-info" }
+com-adarshr-test-logger = { id = "com.adarshr.test-logger", version.ref = "test-logger" }
+com-bmuschko-gradle-cargo = { id = "com.bmuschko.gradle-cargo", version.ref = "gradle-cargo" }
diff --git a/integration-tests/build.gradle b/integration-tests/build.gradle
index 0cd3af35d..c1ad8f2f4 100644
--- a/integration-tests/build.gradle
+++ b/integration-tests/build.gradle
@@ -79,7 +79,7 @@ cargo {
             }
             jvmArgs += ' -Dspring.profiles.active=test -Dfineract.events.external.enabled=true'
             property 'cargo.start.jvmargs', jvmArgs
-            property 'cargo.tomcat.connector.keystoreFile', file("$rootDir/fineract-provider/src/main/resources/keystore.jks")
+            property 'cargo.tomcat.connector.keystoreFile', file("$rootDir/fineract-war/src/main/resources/keystore.jks")
             property 'cargo.tomcat.connector.keystorePass', 'openmf'
             property 'cargo.tomcat.connector.keystoreType', 'JKS'
             property 'cargo.tomcat.httpSecure', true
@@ -169,5 +169,5 @@ tasks.named('compileTestJava') {
 }
 
 // NOTE: Gradle suggested these dependencies
-compileTestJava.dependsOn(':fineract-provider:generateGitProperties', ':fineract-provider:processResources', ':fineract-provider:resolve')
-spotbugsTest.dependsOn(':fineract-provider:generateGitProperties', ':fineract-provider:processResources', ':fineract-provider:resolve')
+compileTestJava.dependsOn(':fineract-war:generateGitProperties', ':fineract-war:processResources')
+spotbugsTest.dependsOn(':fineract-war:generateGitProperties', ':fineract-war:processResources')
diff --git a/integration-tests/dependencies.gradle b/integration-tests/dependencies.gradle
index 95e4b26a5..6cc704e0e 100644
--- a/integration-tests/dependencies.gradle
+++ b/integration-tests/dependencies.gradle
@@ -21,8 +21,8 @@ dependencies {
     // Do NOT repeat dependencies which are ALREADY in implementation or runtimeOnly!
     //
     tomcat 'org.apache.tomcat:tomcat:10.1.55@zip'
-    def providerMainOutput = project(':fineract-provider').extensions.getByType(SourceSetContainer).named('main').get().output
-    testImplementation( providerMainOutput,
+    def warOutput = project(':fineract-war').extensions.getByType(SourceSetContainer).named('main').get().output
+    testImplementation( warOutput,
             project(path: ':fineract-core', configuration: 'runtimeElements'),
             project(path: ':fineract-accounting', configuration: 'runtimeElements'),
             project(path: ':fineract-investor', configuration: 'runtimeElements'),
@@ -31,7 +31,7 @@ dependencies {
             project(path: ':fineract-tax', configuration: 'runtimeElements'),
             project(path: ':fineract-loan', configuration: 'runtimeElements'),
             project(path: ':fineract-savings', configuration: 'runtimeElements'),
-            project(path: ':fineract-provider', configuration: 'runtimeElements'),
+            project(path: ':fineract-war', configuration: 'runtimeElements'),
             project(path: ':fineract-avro-schemas', configuration: 'runtimeElements'),
             project(path: ':fineract-progressive-loan', configuration: 'runtimeElements')
             )
diff --git a/oauth2-tests/dependencies.gradle b/oauth2-tests/dependencies.gradle
index b7dfa1247..911581af5 100644
--- a/oauth2-tests/dependencies.gradle
+++ b/oauth2-tests/dependencies.gradle
@@ -21,8 +21,8 @@ dependencies {
     // Do NOT repeat dependencies which are ALREADY in implementation or runtimeOnly!
     //
     tomcat 'org.apache.tomcat:tomcat:10.1.55@zip'
-    testImplementation( files("$rootDir/fineract-provider/build/classes/java/main/"),
-            project(path: ':fineract-provider', configuration: 'runtimeElements'),
+    testImplementation( files("$rootDir/fineract-war/build/classes/java/main/"),
+            project(path: ':fineract-war', configuration: 'runtimeElements'),
             project(path: ':fineract-security', configuration: 'runtimeElements'),
             'org.junit.jupiter:junit-jupiter-api',
             'org.bouncycastle:bcpkix-jdk18on',
diff --git a/settings.gradle b/settings.gradle
index cc45f2797..ccf93f7aa 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -58,7 +58,6 @@ include ':fineract-command-audit'
 include ':fineract-command-async'
 include ':fineract-command-disruptor'
 include ':fineract-accounting'
-include ':fineract-provider'
 include ':fineract-branch'
 include ':fineract-document'
 include ':fineract-investor'
diff --git a/twofactor-tests/dependencies.gradle b/twofactor-tests/dependencies.gradle
index a91d03851..7e1224050 100644
--- a/twofactor-tests/dependencies.gradle
+++ b/twofactor-tests/dependencies.gradle
@@ -21,8 +21,8 @@ dependencies {
     // Do NOT repeat dependencies which are ALREADY in implementation or runtimeOnly!
     //
     tomcat 'org.apache.tomcat:tomcat:10.1.55@zip'
-    testImplementation( files("$rootDir/fineract-provider/build/classes/java/main/"),
-            project(path: ':fineract-provider', configuration: 'runtimeElements'),
+    testImplementation( files("$rootDir/fineract-war/build/classes/java/main/"),
+            project(path: ':fineract-war', configuration: 'runtimeElements'),
             'org.junit.jupiter:junit-jupiter-api',
             'com.icegreen:greenmail-junit5',
             'org.bouncycastle:bcpkix-jdk18on',
